drivers/docker: fix logging and driver_config declaration

notnoop opened the pull request:
`driver_config` and `logging` stanzas are expected to be a single
optional block as they get deserialized to a single struct, not an
array.

preetapan commented on drivers/docker/config.go:
```diff
@@ -276,7 +276,7 @@ var (
+				"driver_config": hclspec.NewBlock("driver_config", false, hclspec.NewObject(map[string]*hclspec.Spec{
 					"name":    hclspec.NewAttr("name", "string", false),
 					"options": hclspec.NewBlockAttrs("options", "string", false),
 				})),
```
Can we create a parsing unit test that fails before and succeeds after this change?

notnoop replied:
Sadly, we don't have a stable pattern for unit testing task driver config; and most config parsing tests are e2e now AFAIK.

I'll resurrect a branch for unit testing https://github.com/hashicorp/nomad/compare/r-testing-hcl-parsing in a separate PR.

preetapan commented on drivers/docker/config.go:
```diff
@@ -253,7 +253,7 @@ var (
 		"load":               hclspec.NewAttr("load", "string", false),
-		"logging": hclspec.NewBlockSet("logging", hclspec.NewObject(map[string]*hclspec.Spec{
+		"logging": hclspec.NewBlock("logging", false, hclspec.NewObject(map[string]*hclspec.Spec{
 			"type":   hclspec.NewAttr("type", "string", false),
```
Also, does this affect json job parsing that use these attributes?

notnoop replied:
Not sure, but this is currently broken - any use of logging stanza gets:

```
driver start failed: failed to decode driver config: [pos 308]: readContainerLen: Unrecognized descriptor byte: hex: 82, decimal: 130
```

It sounds like we should follow up with json serialization and compatibility testing as a separate work.

notnoop commented:
This was superseded by https://github.com/hashicorp/nomad/pull/5314.  I added tests that reveal the problem and fixed it, as well as config tests for other drivers.
